src/termio/Exec.zig

```diff
@@ -756,8 +757,25 @@ const Subprocess = struct {
         // then we get it ourselves.
         var env = cfg.env;
 
+        var host_resources_dir_buf: [std.fs.max_path_bytes]u8 = undefined;
+        const resources_dir = rd: {
+            const dir = cfg.resources_dir orelse break :rd null;
+
+            // Resolve host OS location of resources directory in flatpak.
+            // Executed programs would not be able to access them otherwise.
+            if (!internal_os.isFlatpak() or FlatpakInfo == void) break :rd dir;
+            var info = FlatpakInfo.init() catch break :rd dir;
+            defer info.deinit();
+
+            break :rd info.resolveAppToHost(dir, &host_resources_dir_buf) catch |err| switch (err) {
+                // Resources directory is not app-relative, ignore
+                error.FlatpakNotAppPath => dir,
+                else => return err,
+            };
+        };
+
```

Comment:
I don't think that we should be adding more apprt-specific behavior here. This is already a very complex function. We should probably figure out a way to have this behavior encapsulated inside the apprt as well.